### PR TITLE
feat: Print action outputs to console

### LINKED ISSUE #1351: Add outputs to screen
I find the pre-europa outputs very useful, just wish you could control which outputs to show.

Europa is well positioned for this, maybe something like:

```cue
engine.#Plan & {
    actions: {
        build: push: engine.#Push & { ... }
        run: push: engine.#Push & { ... }
    }
    outputs: values: {
        "build image": actions.build.push.result
        "run image": actions.run.push.result
    }
}
```
Which would output:
```
Output		Value
build image	registry.example.net/repo:build@sha256:1b2bd4200d734dd3fb819faf21b9352f973f67c55cc8d59addeafc602835461a
run image	registry.example.net/repo:run@sha256:3eaa0b49e5ed5831cfa372cd8f62615012ec87d27401c08f28c7cb1a8bae949f

```

### PR DESCRIPTION
**Pull request by helderco**
Resolves #1351

👀 [Doc preview](https://deploy-preview-2265--devel-docs-dagger-io.netlify.app/1226/handling-outputs)

Signed-off-by: Helder Correia

### CODE REVIEW

**Comment by helderco** [comment]
Getting errors on sops.

**Comment by helderco** [comment]
Update: I'm blocked by the same issue as:
- https://github.com/dagger/dagger/issues/1867

I need to go back to a fix for it in order to move forward:
- https://github.com/dagger/dagger/pull/2024

**Comment by helderco** [comment]
Update: asked for feedback upstream on the blocker: https://github.com/cue-lang/cue/issues/1671

Solution in https://github.com/dagger/dagger/pull/2024 allows to unblock.

👀 Updated [doc preview](https://deploy-preview-2265--devel-docs-dagger-io.netlify.app/1226/handling-outputs)

I tried filling from tasks to a `r.final` but got data races. So approach was to recover computed values and only fill at the end.

Ready to review approach.

File: docs/guides/1228-handling-outputs.md
Comment (marcosnils) [inline docs/guides/1228-handling-outputs.md:57]: nit: "even without a shell"  seems unnecessary to clarify

```suggestion
You can also save the output to a file using the `--output` flag. Let's do it in _yaml_ this time:
```

**Review by marcosnils — COMMENTED** [review]
Just minor nit comment. Only reviewed the `.md` instructions and they LGTM!

**Comment by aluzzardi** [comment]
Haven't looked in depth yet but looks good!

Q: Aren't the outputs polluted by random fields, or is there some logic to clean that up? (I looked at the logic of `isOutput` but couldn't find anything related to that).

For instance, for a `netlify.#Deploy`, shouldn't we see a bunch of noise in `deploy.container` (generated by `bash.#Run`)?

**Comment by helderco** [comment]
> Q: Aren't the outputs polluted by random fields, or is there some logic to clean that up? (I looked at the logic of `isOutput` but couldn't find anything related to that).
> 
> For instance, for a `netlify.#Deploy`, shouldn't we see a bunch of noise in `deploy.container` (generated by `bash.#Run`)?

No because we're using only top-level fields of an action, not going deep. The `netlify.#Deploy` action is a good example as it correctly exports to the top-level what outputs are of interest:

```cue
	// URL of the deployed site
	url: container.export.files."/netlify/url"

	// URL of the latest deployment
	deployUrl: container.export.files."/netlify/deployUrl"

	// URL for logs of the latest deployment
	logsUrl: container.export.files."/netlify/logsUrl"
```

Which produces the output you can see [here](https://deploy-preview-2265--devel-docs-dagger-io.netlify.app/1226/handling-outputs/):

```console
➜  APP_NAME=dagger-todo dagger do deploy
[✔] actions.deps
[✔] actions.test.script
[✔] client.env
[✔] actions.build.run.script
[✔] actions.deploy.container.script
[✔] client.filesystem."./".read
[✔] actions.deploy
[✔] actions.test
[✔] actions.build.run
[✔] actions.build.contents
[✔] actions.deploy.container
[✔] client.filesystem."./_build".write
[✔] actions.deploy.container.export
Field      Value
url        "https://dagger-todo.netlify.app"
deployUrl  "https://62698983ffe8661d60613431--dagger-todo.netlify.app"
logsUrl    "https://app.netlify.com/sites/dagger-todo/deploys/62698983ffe8661d60613431"
```

So you can create a wrapper action to control which outputs you want, as you can see in this example: https://deploy-preview-2265--devel-docs-dagger-io.netlify.app/1226/handling-outputs/#controlling-the-output

**Comment by aluzzardi** [comment]
> The netlify.#Deploy action is a good example as it correctly exports to the top-level what outputs are of interest

Yes, I noticed the output is perfect, that's what got me wondering on what's the wizardry behind that makes it look good :)

> No because we're using only top-level fields of an action

How do we define top-level fields? For instance:

- `web` contains `build` and `deploy` (`build is yarn and `deploy` is netlify)
- `web.deploy` is netlify
- `web.deploy.container` is an internal field of netlify (`bash.#Run`)

`web`, `deploy` and `container` are actions in this case.

What would end up in outputs?

**Comment by helderco** [comment]
> What would end up in outputs? Is it going to be the outputs of `web` only?

Yes 🙂

**Comment by aluzzardi** [comment]
Ohh, clever! Makes sense -- this is the perfect counter part of `--input`s, top-level only.

**Review by aluzzardi — APPROVED** [review]
Design & Code LGTM

**Review by shykes — APPROVED** [review]
Can’t wait to play with this!

I’m sure we’ll find rough edges but we can deal with them as we go.

Thanks @helderco !

**Review by marcosnils — APPROVED** [review]

**Pull request merged**
